File: chessgame-api/src/main/java/git/dimitrikvirik/chessgameapi/model/domain/Game.java
package git.dimitrikvirik.chessgameapi.model.domain;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;


@Entity
@Data
@NoArgsConstructor
public class Game {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;


    @ManyToOne
    private UserAccount whitePlayer;

    @ManyToOne
    private UserAccount blackPlayer;

    @ManyToOne
    private UserAccount winnerPlayer;

    @OneToMany(mappedBy = "game")
    private List<GameHistory> history;

}

File: chessgame-api/src/main/kotlin/git/dimitrikvirik/chessgameapi/model/domain/Game.kt
package git.dimitrikvirik.chessgameapi.model.domain

import javax.persistence.*

@Entity
@Table(name = "game")
class Game(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) var id: Int? = null,
    @ManyToOne
    val whitePlayer: UserAccount? = null,
    @ManyToOne
    val blackPlayer: UserAccount? = null,
    @ManyToOne
    val winnerPlayer: UserAccount? = null,
    @OneToMany(mappedBy = "game")
    val history: List<GameHistory>? = null
)